src/lib/converter/factories/declaration.ts

```diff
@@ -45,7 +63,8 @@ export function createDeclaration(context: Context, node: ts.Node, kind: Reflect
 
     // Test whether the node is exported
     let isExported: boolean;
-    if (container.kindOf([ReflectionKind.Module, ReflectionKind.ExternalModule])) {
+    if (container.kindOf([ReflectionKind.Module, ReflectionKind.ExternalModule]) || considerInheritedExportKinds.indexOf(kind) === -1
+        || node.kind === ts.SyntaxKind.VariableDeclaration || node.kind === ts.SyntaxKind.FunctionDeclaration) {
```

Comment:
Can you help me understand this line a bit? Part of the confusion is the existing code but I'm not following why this logic is added here.

From what I can see these lines are turning `isExported` off not just for modules but also for kinds that inherit their export status and for variable declarations and for function declarations. Then below `isExported` can be marked true by a modifier flag.

Because we are reassigning the `isExported` flag multiple times it is unclear what each step is checking. Can you give a bit higher level description of the logic here?

Reply:
Thanks for your detailed reply.

I was unsure about removing the original line ``container.kindOf([ReflectionKind.Module, ReflectionKind.ExternalModule])``. I didn't want to destroy something but I think it can be removed.

The code is directly related to #472 (see discussion there). If you have merged types/namespaces the container kind sometimes isn't a module but the type (class/interface...). So any sub-declaration inherits the class's export state. But that's not true for variables/functions that are defined within the merged namespace.

So my point here is that it is in fact the declaration's kind that specifies if the exported state could be inherited, not the container's kind.

The second line (VariableDeclaration/FunctionDeclaration) addresses the issue that variables/functions inside the namespace are treated as static properties/methods (kind is Property/Method that actually would inherit the export state of the class).

See [converter/nodes/variable.ts](https://github.com/TypeStrong/typedoc/blob/3102b2b2ce6936410c05a56f517b1f3fa96e4a5f/src/lib/converter/nodes/variable.ts#L62) or [converter/nodes/function.ts](https://github.com/TypeStrong/typedoc/blob/3102b2b2ce6936410c05a56f517b1f3fa96e4a5f/src/lib/converter/nodes/function.ts#L29)

Generally this part of the code is pretty confusing (I came upon it as well) and surely could be done better. But as far as I can say the last reassignment just handles the "export" keyword, so if it is explicitely exported, the element is in fact exported.